contracts/token/ERC721/extensions/draft-ERC721Royalty.sol

```diff
@@ -3,78 +3,125 @@
 
 pragma solidity ^0.8.0;
 
+import "../ERC721.sol";
 import "./draft-IERC721Royalty.sol";
-import "../../../utils/introspection/ERC165Storage.sol";
-import "hardhat/console.sol";
+import "../../../utils/introspection/ERC165.sol";
 
 /**
  * @dev Implementation of the ERC721 Royalty extension allowing royalty information to be stored and retrieved, as defined in
  * https://eips.ethereum.org/EIPS/eip-2981[EIP-2981].
  *
- * Adds the {_setTokenRoyalty} methods to set the token royalty information, and {_setRoyalty} method to set a global
+ * Adds the {_setTokenRoyalty} methods to set the token royalty information, and {_setGlobalRoyalty} method to set a global
  * royalty information.
  *
  * _Available since v4.5._
  */
-abstract contract ERC721Royalty is IERC721Royalty, ERC165Storage {
+abstract contract ERC721Royalty is IERC721Royalty, ERC721 {
     struct RoyaltyInfo {
         address receiver;
-        uint256 royaltyAmount;
+        uint256 royaltyFraction;
     }
 
-    RoyaltyInfo private _royaltyInfo;
-    mapping(uint256 => RoyaltyInfo) private _tokenRoyalty;
+    RoyaltyInfo private _globalRoyaltyInfo;
+    mapping(uint256 => RoyaltyInfo) private _tokenRoyaltyInfo;
 
-    /*
+    /**
      * @dev Sets tokens royalties
      *
      * Requirements:
      * - `tokenId` must be already mined.
-     * - `recipient` cannot be the zero address.
-     * - `value` must indicate the percentage value using two decimals.
+     * - `receiver` cannot be the zero address.
+     * - `fraction` must indicate the percentage fraction using two decimals.
      */
     function _setTokenRoyalty(
         uint256 tokenId,
-        address recipient,
-        uint256 value
+        address receiver,
+        uint256 fraction
     ) internal virtual {
-        require(value < 100, "ERC2981: Royalty percentage is too high");
-        require(recipient != address(0), "ERC2981: Invalid recipient");
+        require(fraction > 0, "ERC2981: Royalty percentage is too low");
```

Comment:
I think we need to allow 0. It would be the only workaround to have royalties globally but disable them per-token.

Reply:
But you could call the deleteTokenRoyalty function